src/Sylius/Component/Core/Model/Product.php

```diff
@@ -355,4 +355,16 @@ public function getImage()
 
         return $this->getFirstVariant()->getImage();
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getImages()
+    {
+        if (null === $this->getFirstVariant()) {
+            return new ArrayCollection();
```

Comment:
I'm not sure if I'm reading this correctly but from the `ProductInterface` it looks like the `firstVariant()` is supposed to return an `ImageInterface` but in the case where there is no first variant we return an `ArrayCollection`. Is this correct? If so is it possible to convert the result of `getImages()` to an `ArrayCollection`, or possibly return a null image interface.

Eg:

```
if (null === $this->getFirstVariant() {
    return new EmptyImageInterface();
}

// etc ...
```